cmake/Modules/FunctionQbtCXXCompilerAndModeCheck.cmake

```diff
@@ -0,0 +1,19 @@
+function(qbt_cxx_compiler_and_mode_check _min_std)
```

Comment:
Really I don't like this function. I would put all the necessary code directly in CMakeLists.txt.

Reply:
@glassez

This type of code seems like the ideal candidate to be factored out to a function. It only does one thing, which is useful, but that most people don't care about the details (so the full code should not clutter the main script): it ensures qBittorrent will be built with the minimum required C++ version.

I decided to change the name to `qbt_minimum_cxx_mode_check` to simplify it/make it more obvious, though.